### PR TITLE
KMS-4100 bse preview should be lti launched

### PR DESCRIPTION
**Pull request by muli**

### CODE REVIEW

**Comment by gonenradai** [comment]
does that cover both Atto and TinyMCE?

**Comment by muli** [comment]
Yup.

**Comment by alon1980** [comment]
@muli whats the status here?

**Comment by muli** [comment]
@alon1980 - waiting for Gonen's review.
You can review too if you want :)

File: local/kaltura/service.php [hunk 1]
```diff
@@ -90,6 +90,7 @@
 
 $PAGE->set_url($serviceurl);
 $PAGE->set_context(context_system::instance());
+$previewltilaunchurl = new moodle_url('/local/kaltura/bsepreview_ltilaunch.php?playurl=' . urlencode($url));
```
Comment (alon1980) [inline local/kaltura/service.php:93]: camelCasing

File: local/kaltura/service.php [hunk 2]
```diff
@@ -98,6 +99,7 @@
     'title' => $title,
     'metadata' => $metadata,
     'editor' => $editor,
+    'previewltilauncher' => $previewltilaunchurl->out(),
```
Comment (alon1980) [inline local/kaltura/service.php:102]: same

Comment (alon1980) [inline local/kaltura/service.php:102]: what is out() ? like echo?

**Comment by alon1980** [comment]
you also have some conflicts here

File: local/kaltura/service.php (hunk 2 above)
Comment (muli) [inline local/kaltura/service.php:102]: it's a `moodle_url` class method that outputs the full URL. it doesn't echo.

File: local/kaltura/service.php (hunk 1 above)
Comment (muli) [inline local/kaltura/service.php:93]: i usually do camel casing, however, if you look around Moodle's code (and our plugin's as well) you'd notice that the code is not camel cased, for the most part. it's more of a C-style code. so i kept on with it.

File: local/kaltura/service.php (hunk 2 above)
Comment (alon1980) [inline local/kaltura/service.php:102]: hate it but ok i guess.

**Comment by alon1980** [comment]
LGTM

File: local/kaltura/service.php (hunk 2 above)
Comment (muli) [inline local/kaltura/service.php:102]: why do you hate it?
it's a pretty standard in Moodle...

Comment (alon1980) [inline local/kaltura/service.php:102]: because its hard to read variables thisiswhy (or better yet - thisIsWhy)

Comment (muli) [inline local/kaltura/service.php:102]: oh... we responded to each other on different subjects.
I thought that your hate was towards the `moodle_url` thing. :)

**Pull request merged**